src/main/java/org/spongepowered/common/data/processor/data/entity/TargetLivingDataProcessor.java

```diff
@@ -0,0 +1,96 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.entity;
+
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.boss.EntityWither;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableTargetLivingData;
+import org.spongepowered.api.data.manipulator.mutable.entity.TargetLivingData;
+import org.spongepowered.api.data.value.immutable.ImmutableListValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.entity.living.Living;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeTargetLivingData;
+import org.spongepowered.common.data.processor.common.AbstractEntitySingleDataProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class TargetLivingDataProcessor extends
+        AbstractEntitySingleDataProcessor<EntityLivingBase, List<Living>, ImmutableListValue<Living>, TargetLivingData, ImmutableTargetLivingData> {
+
+    public static int MAX_TARGET_INDEX = 2;
+
+    public TargetLivingDataProcessor() {
+        super(EntityLivingBase.class, Keys.TARGETS);
+    }
+
+    @Override
+    protected boolean set(EntityLivingBase entity, List<Living> value) {
+        boolean hasSet = false;
+        for (int i = 0; i < MAX_TARGET_INDEX; i++) {
+            if (entity instanceof EntityWither) {
+                ((EntityWither) entity).func_82211_c(i, value.size() > i ? ((EntityLivingBase) value.get(i)).getEntityId() : 0);
+                hasSet = true;
+            }
+        }
+        return hasSet;
+    }
+
+    @Override
+    protected Optional<List<Living>> getVal(EntityLivingBase entity) {
+        List<Living> values = new ArrayList<>();
+        if (entity instanceof EntityWither) {
+            EntityWither wither = (EntityWither) entity;
```

Comment:
If you're focusing on `EntityWither`, why not just make it part of the processor requirement?